jina/peapods/pods/compoundpod.py

```diff
@@ -232,9 +225,10 @@ def _set_replica_args(
             result.append(_args)
         return result
 
-    def rolling_update(self):
-        """
-        Update all pods of this compound pod.
+    def rolling_update(self, dump_path):
```

Comment:
type hinting, this should be an `optional` parameter defaulting to `None`. This way we can also test without `dump_path` (Better split of infra/functional code)

Reply:
I actually don't see why it should be done here, as it kinda implies you could do rolling update without a path. It should be passed a `None` in tests where we don't care that it's correct. But by default, it should not allow users to start a query flow without a dump paht

Reply:
if in tests will be passed to None is because it is accepting.

The concern that this should not work for Query Flows is handled somewhere else.

Here the CompoundPod is just asked to restart its replicas.

Reply:
It works to just pass `None` in the rolling update test, so no problem. It works there because we don't care about whatever dummy indexer receives it and doesn't use it.

It should not be "invisible", though, as users might think it is truly optional, when it isn't.

Reply:
I would even generallize and have a set of params to update for the Pod. So that this is less linked to the actual functionality.

But no big deal, we can change later. But since the parametrr accepts None, the thpe hint should be there.

The point is to keep a distinction about the functionality it offers and the infrastructure used for it.

Reply:
Added type annotations, but I think we should keep it as is otherwise. The distinction is only there in the test. You wouldn't use it without a proper dump to reload.